Harmony/Public/Harmony.cs

```diff
@@ -210,52 +210,62 @@ public static MethodInfo ReversePatch(MethodBase original, HarmonyMethod standin
 		///
-		public void UnpatchAll(string harmonyID = null)
+		public static void UnpatchAll()
 		{
-			if (harmonyID == null)
+			if (HarmonyGlobalSettings.DisallowGlobalUnpatchAll)
 			{
 				if (HarmonyGlobalSettings.DisallowGlobalUnpatchAll)
 				{
-					Logger.Log(Logger.LogChannel.Warn, () => "UnpatchAll has been called with harmonyID=null AND DisallowGlobalUnpatchAll=true. " +
-					                                         "If you want to only unpatch patches created by this instance (" + Id + "), use UnpatchSelf() instead.");
+					Logger.Log(Logger.LogChannel.Warn, () => "UnpatchAll has been called AND DisallowGlobalUnpatchAll=true. " +
+					                                         "Skipping execution of UnpatchAll");
 					return;
 				}
-
-				Logger.Log(Logger.LogChannel.Warn, () => "UnpatchAll has been called with harmonyID=null - This will remove ALL HARMONY PATCHES. " +
-				                                         "If you want to only unpatch patches created by this instance (" + Id + "), use UnpatchSelf() instead.");
 			}
 
-			bool IDCheck(Patch patchInfo)
-			{
-				return harmonyID is null || patchInfo.owner == harmonyID;
-			}
+			Logger.Log(Logger.LogChannel.Warn, () => "UnpatchAll has been called - This will remove ALL HARMONY PATCHES.");
 
-			var originals = GetAllPatchedMethods().ToList(); // keep as is to avoid "Collection was modified"
-			foreach (var original in originals)
-			{
-				var hasBody = original.HasMethodBody();
-				var info = GetPatchInfo(original);
-				if (hasBody)
-				{
-					info.Postfixes.DoIf(IDCheck, patchInfo => Unpatch(original, patchInfo.PatchMethod));
-					info.Prefixes.DoIf(IDCheck, patchInfo => Unpatch(original, patchInfo.PatchMethod));
-				}
-				info.ILManipulators.DoIf(IDCheck, patchInfo => Unpatch(original, patchInfo.PatchMethod));
-				info.Transpilers.DoIf(IDCheck, patchInfo => Unpatch(original, patchInfo.PatchMethod));
-				if (hasBody)
-					info.Finalizers.DoIf(IDCheck, patchInfo => Unpatch(original, patchInfo.PatchMethod));
-			}
+			PatchFunctions.UnpatchConditional(_ => true);
 		}
 
-		/// <summary>Unpatches all methods that were patched by this Harmony instance's ID. Unpatching is done by repatching methods without patches of this instance.</summary>
+		/// <summary>Unpatches methods by patching them with zero patches. Fully unpatching is not supported. Be careful, unpatching is global</summary>
+		/// <param name="harmonyID">The Harmony ID to restrict unpatching to a specific Harmony instance. Whether this parameter is actually optional is determined by the <see cref="HarmonyGlobalSettings.DisallowGlobalUnpatchAll"/> global flag</param>
+		/// <remarks>When <see cref="HarmonyGlobalSettings.DisallowGlobalUnpatchAll"/> is set to true, the execution of this method will be skipped when no <paramref name="harmonyID"/> is specified.</remarks>
 		///
-		public void UnpatchSelf()
+		[Obsolete("Use UnpatchSelf() to unpatch the current instance", true)]
```

Comment:
There should be an additional message indicating that the original functionality has been moved to the new static methods

Reply:
Changed.